api/test/unit/utils/test_photon.py

```diff
@@ -216,76 +218,133 @@ def raise_exc(*args, **kwargs):
     yield do
 
 
-def test_get_timeout_no_existing_cache_key(
-    capture_exception, setup_requests_get_exception, redis
-):
-    exc = requests.ReadTimeout()
-    setup_requests_get_exception(exc)
+@pook.on
+def test_get_successful_records_response_code(mock_image_data, redis):
+    (
+        pook.get(PHOTON_URL_FOR_TEST_IMAGE)
+        .params(
+            {
+                "w": settings.THUMBNAIL_WIDTH_PX,
+                "quality": settings.THUMBNAIL_QUALITY,
+            }
+        )
+        .header("User-Agent", UA_HEADER)
+        .header("Accept", "image/*")
+        .reply(200)
+        .body(MOCK_BODY)
+        .mock
+    )
 
-    with pytest.raises(UpstreamThumbnailException):
-        photon_get(TEST_IMAGE_URL)
+    photon_get(TEST_IMAGE_URL)
+    month = get_monthly_timestamp()
+    assert redis.get(f"thumbnail_response_code:{month}:200") == b"1"
+    assert (
+        redis.get(
+            f"thumbnail_response_code_by_domain:subdomain.example.com:{month}:200"
+        )
+        == b"1"
+    )
 
-    capture_exception.assert_called_once_with(exc)
 
-    key = f"{settings.THUMBNAIL_TIMEOUT_PREFIX}subdomain.example.com"
-    assert redis.get(key) == b"1"
+alert_count_params = pytest.mark.parametrize(
+    "count_start, should_alert",
```

Comment:
Would count alert be more appropriate? Seeing how the alert count maps 1-1 with the event count till 100, stays there for 900 more, then adds 1 for every thousand?
```suggestion
    "count_start, count_alert",
```

Reply:
Hmm, I think I find `count_alert` a bit more confusing to use here 😅 The second value describes whether or not the next exception should be alerted to Sentry, rather than an actual count of how many alerts are made.